app/models/spreadsheet_service.rb

```diff
@@ -0,0 +1,31 @@
+require 'google/apis/sheets_v4'
+
+class SpreadsheetService
+  def self.get_data(spreadsheet_id)
+    service = Google::Apis::SheetsV4::SheetsService.new
+    service.key = ENV['GOOGLE_SHEET_API_KEY']
+
+    range = SpreadsheetService.get_range(spreadsheet_id)
+    begin
+      response = service.get_spreadsheet_values(spreadsheet_id, range)
+    rescue Exception => e
+      raise ActionController::BadRequest.new(), e.message()
+    end
+
+    response.values
+  end
+
+  def self.get_range(spreadsheet_id)
+    service = Google::Apis::SheetsV4::SheetsService.new
+    service.key = ENV['GOOGLE_SHEET_API_KEY']
```

Comment:
Let's take a look at how other configurations are used on the app.

Rails has the `Settings` global object that allows users to configure the app via different mechanisms (the different environments' yml files, environment variables, etc). Instead of only supporting environment variables, let's use that. And we avoid having direct `ENV` access from the business code - which is harder to track.